lib/components/ConnectModal/index.tsx

```diff
@@ -50,44 +76,108 @@ export const ConnectModal = (props: Props) => {
       } else if (params.wallet === "brave" && connect) {
         await connect("brave");
       }
-      setShowModal(false);
+      toggleModal && toggleModal();
       setStep("connect");
       props.onClose && props.onClose();
     } catch (e: any) {
       console.error(e);
       setStep("error");
     }
   };
-
-  return (
-    <>
-      {isConnected && address ? (
-        <ButtonPrimary
-          label={concatAddress(address)}
-          onClick={disconnect}
-          variant={buttonVariant}
-        />
-      ) : (
-        <ButtonPrimary
-          label={props.buttonText}
-          onClick={() => {
-            setShowModal(true);
-          }}
-          variant={buttonVariant}
-          className={props.buttonClassName}
-        />
-      )}
-
-      <ConnectContent
-        showModal={showModal}
-        handleConnect={handleConnect}
-        setShowModal={setShowModal}
-        step={step}
-        setStep={setStep}
-        errorMessage={props.errorMessage}
-        torusText={props.torusText}
-        titles={props.titles}
+  return props.showModal ? (
```

Comment:
I would just early exist one line above. keeps it more flat and readable
```
if (!props.showModal) return null
```

Reply:
updated